batch/plugins/org.jboss.tools.batch.ui/src/org/jboss/tools/batch/ui/quickfixes/BatchQuickFixGenerator.java

```diff
@@ -0,0 +1,262 @@
+/*******************************************************************************
+ * Copyright (c) 2015 Red Hat, Inc.
+ * Distributed under license by Red Hat, Inc. All rights reserved.
+ * This program is made available under the terms of the
+ * Eclipse Public License v1.0 which accompanies this distribution,
+ * and is available at http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     Red Hat, Inc. - initial API and implementation
+ ******************************************************************************/
+package org.jboss.tools.batch.ui.quickfixes;
+
+import org.eclipse.core.resources.IFile;
+import org.eclipse.core.resources.IMarker;
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.jdt.core.IJavaElement;
+import org.eclipse.jdt.core.IJavaProject;
+import org.eclipse.jdt.core.JavaModelException;
+import org.eclipse.jdt.ui.text.java.IJavaCompletionProposal;
+import org.eclipse.jface.text.BadLocationException;
+import org.eclipse.jface.text.IDocument;
+import org.eclipse.jface.text.Position;
+import org.eclipse.jface.text.source.Annotation;
+import org.eclipse.ui.IMarkerResolution;
+import org.eclipse.ui.IMarkerResolutionGenerator2;
+import org.eclipse.ui.part.FileEditorInput;
+import org.eclipse.ui.texteditor.DocumentProviderRegistry;
+import org.eclipse.ui.texteditor.IDocumentProvider;
+import org.eclipse.wst.sse.ui.internal.reconcile.TemporaryAnnotation;
+import org.jboss.tools.batch.core.BatchArtifactType;
+import org.jboss.tools.batch.internal.core.validation.BatchValidator;
+import org.jboss.tools.batch.ui.BatchUIPlugin;
+import org.jboss.tools.common.EclipseUtil;
+import org.jboss.tools.common.quickfix.IQuickFix;
+import org.jboss.tools.common.quickfix.IQuickFixGenerator;
+import org.jboss.tools.common.refactoring.MarkerResolutionUtils;
+import org.jboss.tools.common.validation.ValidationErrorManager;
+
+public class BatchQuickFixGenerator implements IMarkerResolutionGenerator2, IQuickFixGenerator{
+	private static final String XML_EXTENSION = "xml"; //$NON-NLS-1$
+
+	@Override
+	public IMarkerResolution[] getResolutions(IMarker marker) {
```

Comment:
6 methods (out of 10) of this class and methods in CDIProblemMarkerResolutionGenerator class are identical. Looks like we need some abstract class which will be inherited by BatchQuickFixGenerator and CDIProblemMarkerResolutionGenerator.